ratatui-widgets/src/mascot.rs

```diff
@@ -0,0 +1,196 @@
+//! A Ratatui mascot widget
+//!
+//! The mascot takes 32x32 cells and is rendered using half block characters.
+use itertools::Itertools;
+use ratatui_core::{buffer::Buffer, layout::Rect, style::Color, widgets::Widget}; // tuples();
+
+const RATATUI_MASCOT: &str = indoc::indoc! {"
+                   hhh
+                 hhhhhh
+                hhhhhhh
+               hhhhhhhh
+              hhhhhhhhh
+             hhhhhhhhhh
+            hhhhhhhhhhhh
+            hhhhhhhhhhhhh
+            hhhhhhhhhhhhh     ██████
+             hhhhhhhhhhh    ████████
+                  hhhhh ███████████
+                   hhh ██ee████████
+                    h █████████████
+                ████ █████████████
+               █████████████████
+               ████████████████
+               ████████████████
+                ███ ██████████
+              ▒▒    █████████
+             ▒░░▒   █████████
+            ▒░░░░▒ ██████████
+           ▒░░▓░░░▒ █████████
+          ▒░░▓▓░░░░▒ ████████
+         ▒░░░░░░░░░░▒ ██████████
+        ▒░░░░░░░░░░░░▒ ██████████
+       ▒░░░░░░░▓▓░░░░░▒ █████████
+      ▒░░░░░░░░░▓▓░░░░░▒ ████  ███
+     ▒░░░░░░░░░░░░░░░░░░▒ ██   ███
+    ▒░░░░░░░░░░░░░░░░░░░░▒ █   ███
+    ▒░░░░░░░░░░░░░░░░░░░░░▒   ███
+     ▒░░░░░░░░░░░░░░░░░░░░░▒ ███
+      ▒░░░░░░░░░░░░░░░░░░░░░▒ █"
+};
+
+const RAT: char = '█';
+const HAT: char = 'h';
+const EYE: char = 'e';
+const TERM: char = '░';
+const TERM_BORDER: char = '▒';
+const TERM_CURSOR: char = '▓';
+
+/// State for the mascot's eye
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum MascotEye {
+    /// The default eye color
+    #[default]
+    Default,
+
+    /// The red eye color
+    Red,
+}
```

Comment:
nit: this seems like a bit of an ambiguous name (but I don't know a better one). Default is one of those names that tends not to be used as a variant name. Would Normal (or open) and Red (or blink) make sense as variant names?

This is very much in the small irrelevant change bucket, so if you're happy with these I'm fine with it and will leave it until some obvious better name hits my subconscious.

Reply:
Probably should be `MascotEyeColor`? 

And maybe `Transparent` instead of `Default` as the default variant?

Reply:
I'd avoid transparency, because it makes the rendering dependent on what is in the buffer before rendering. This is ok as an option, but not as a default. Similar comment on https://github.com/ratatui/ratatui/pull/1584#discussion_r1896714671

Reply:
`MascotEyeColor` sounds better to me. Alternatively, we can simply go without an enum and let the user set the `Color` via `set_eye_color`. e.g. `RatatuiMascot::new().set_eye_color(Color::Red);`

Reply:
I think keeping a bit of control over the logo and how it can be presented is a good idea rather than making this open. That includes specifics of colors etc.

Reply:
Sure, that makes sense.